openeo_processes_dask/process_implementations/cubes/aggregate.py

```diff
@@ -109,11 +123,146 @@ def aggregate_temporal_period(
     )
 
 
+def _aggregate_geometry(
+    data: RasterCube,
+    geom,
+    transform,
+    reducer: Callable,
+):
+    data_dims = list(data.dims)
+    y_dim = data.openeo.y_dim
+    x_dim = data.openeo.x_dim
+    t_dim = data.openeo.temporal_dims
+    b_dim = data.openeo.band_dims
+
+    y_dim_size = data.sizes[y_dim]
+    x_dim_size = data.sizes[x_dim]
+
+    if len(t_dim) == 0:
+        t_dim = None
+    else:
+        t_dim = t_dim[0]
+    if len(b_dim) == 0:
+        b_dim = None
+    else:
+        b_dim = b_dim[0]
+
+    # Create a GeoSeries from the geometry
+    geo_series = gpd.GeoSeries(geom)
+
+    # Convert the GeoSeries to a GeometryArray
+    geometry_array = geo_series.geometry.array
+
+    mask = rasterio.features.geometry_mask(
+        geometry_array, out_shape=(y_dim_size, x_dim_size), transform=transform
+    )
+
+    if t_dim is not None:
+        mask = np.expand_dims(mask, axis=data_dims.index(t_dim))
+    if b_dim is not None:
+        mask = np.expand_dims(mask, axis=data_dims.index(b_dim))
+
+    masked_data = data * mask
+    del mask, data
+    gc.collect()
+
+    positional_parameters = {"data": 0}
+
+    stat_within_polygon = masked_data.reduce(
+        reducer,
+        axis=(data_dims.index(y_dim), data_dims.index(x_dim)),
+        keep_attrs=True,
+        ignore_nodata=True,
+        positional_parameters=positional_parameters,
+    )
+    result = stat_within_polygon.values
+
+    del masked_data, stat_within_polygon
+    gc.collect()
+
+    return result.T
+
+
 def aggregate_spatial(
     data: RasterCube,
-    geometries: VectorCube,
+    geometries,
     reducer: Callable,
-    target_dimension: str = "result",
-    **kwargs,
+    chunk_size: int = 2,
 ) -> VectorCube:
-    raise NotImplementedError
+    t_dim = data.openeo.temporal_dims
+    t_dim_name = t_dim[0]
```

Comment:
It would make sense to also add 

t_dim_name = None if len(t_dim) == 0 else t_dim[0] 

otherwise this line fails when the temporal dimension is not available. This would also need to be considered in Line 231 and 243.